raspberry/main.py

```diff
@@ -0,0 +1,369 @@
+# NOTE: It is encouraged to write all the functions that are need to be scheduled in this document,
+# that way all the shared variables can be accessed by the functions without needing to explicitly pass them
+# and it will make handling shared resources easy.
+
+import numpy as np  # for matrix based calculations
+import pandas as pd  # pandas for storing cached data into csv files
+
+from time import time_ns  # function to get system-up time in nano seconds
+from datetime import datetime  # to get date-time stamps for naming the csv files
+from math import pi, atan2, sqrt  # math functions for calculations
+
+from bzzz.controllers.altitude_LQR import LQR  # altitude hold controller
+from bzzz.estimators.altitude_Kalman_filter import KalmanFilter
+from bzzz.scheduler import Scheduler
+from bzzz.sensors.time_of_flight_sensor import TimeOfFlightSensor
+from bzzz.read_sbus import RC  # for radio data receving, encoding and sending to ESP
+
+
+# NOTE: The scheduler supports both multi-threading and time-based function calling
+# Although threading guarantees consistent function call rates, the actual process handeling is not done
+# within the python environment which could cause unwanted behaviour.
+
+if __name__ == '__main__':
+    # sampling frequency of KF and LQR
+    sampling_frequency = 50
+    # caching configuration
+    enable_caching = [True]
+    enable_save_cache_to_csv = [True and enable_caching[0]]
+    enable_printing_cache_to_screen = [True and enable_caching[0]]
+
+    # objects declaration
+    kf = KalmanFilter(sampling_frequency=sampling_frequency,
+                      initial_Tt=0,
+                      x_tilde_0=np.array([[0], [0], [10], [-9.81]]),
+                      P_0=np.diagflat([1, 1, 1, 0.01]),
+                      cache_values=True)
+    lqr = LQR(sampling_frequency=sampling_frequency,
+              initial_alpha_t=10,
+              initial_beta_t=-9.81)
+    tof = TimeOfFlightSensor(use_sleep=-1,
```

Comment:
use_sleep can only be `True` or `False` (not -1)

Reply:
Done!